linked_list.py

```diff
@@ -33,48 +23,50 @@ def search(self, val):
                 val: String value associated with the val property of Node
         """
         current = self.head
-        exists = False
-        while current and exists is False:
-            if current.get_info() == val:
-                exists = True
+        while current:
+            if current.val == val:
+                break
             else:
```

Comment:
the `else` is redundant.  if you enter the `if` clause, you'll never make it to the `else`.